spark/src/main/scala/geotrellis/spark/io/SparkLayerCopier.scala

```diff
@@ -0,0 +1,36 @@
+package geotrellis.spark.io
+
+import geotrellis.spark.io.json._
+import geotrellis.spark._
+import geotrellis.spark.io.index.KeyIndex
+import org.apache.avro.Schema
+import org.apache.spark.rdd.RDD
+import spray.json._
+import scala.reflect._
+
+class SparkLayerCopier[Header: JsonFormat, K: Boundable: JsonFormat: ClassTag, V: ClassTag, Container]
+  (attributeStore: AttributeStore[JsonFormat],
+   layerReader: FilteringLayerReader[LayerId, K, Container],
+   layerWriter: Writer[LayerId, Container with RDD[(K, V)]])
+  (implicit val cons: ContainerConstructor[K, V, Container], containerEv: Container => Container with RDD[(K, V)]) extends LayerCopier[LayerId] {
+
+  def copy(from: LayerId, to: LayerId): Unit = {
```

Comment:
Is it possible to rewrite this function to: 

```scala
def copy(from: LayerId, to: LayerId): Unit = {
    if (!attributeStore.layerExists(from)) throw new LayerNotFoundError(from)
    if (attributeStore.layerExists(to)) throw new LayerExistsError(to)    
    try {
      layerWriter.write(to, layerReader.read(from))
    } catch { 
      case e: Exception => new LayerCopyError(from, to).initCause(e)
    }
    attributeStore.copy(from, to)
  }
```

For Accumulo, seems there no significant difference. But for other backends definitely lots of difference. 
We just need to decide, should we provide `inspark` copy, or not? If No, I will use `attributeStore.copy` function.
The pros of this `SparkLayerCopier` now that it is generic and can be used with any backend.